sqlglot/tokens.py

```diff
@@ -1203,6 +1204,13 @@ def _extract_string(self, delimiter: str, escapes=None) -> str:
                 if self._end:
                     raise TokenError(f"Missing {delimiter} from {self._line}:{self._start}")
 
+                if self.ESCAPE_SEQUENCES and self._peek and self._char in self.STRING_ESCAPES:
+                    escaped_sequence = self.ESCAPE_SEQUENCES.get(self._char + self._peek)
+                    if escaped_sequence:
+                        self._advance(2)
```

Comment:
should this be len(escaped_sequence)

Reply:
It should be incorrect here because otherwise it'd be`len('\n') == 1`; the `2` comes from the `self._char + self._peek` part, i.e. skip these two characters that were condensed into one.

This logic doesn't take into account escape sequences of different lengths, so we'd need to do another refactor to support them if any dialect has them

Reply:
ok